File: Desarrollos AddOn SAP B1/Samples/COM UI/CSharp/20.PrintEvent/PrintEvent.cs
//  SAP MANAGE UI API 2007 SDK Sample
//****************************************************************************
//
//  File:      PrintEvent.cs
//
//  Copyright (c) SAP MANAGE
//
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//
// BEFORE STARTING:
// 1. Add reference to the "SAP Business One UI API"
// 2. Insert the development connection string to the "Command line argument"
//-----------------------------------------------------------------
// 1.
//    a. Project->Add Reference...
//    b. select the "SAP Business One UI API 2007" From the COM folder
//
// 2.
//     a. Project->Properties...
//     b. choose Configuration Properties folder (place the arrow on Debugging)
//     c. place the following connection string in the 'Command line arguments' field
// 0030002C0030002C00530041005000420044005F00440061007400650076002C0050004C006F006D0056004900490056
//
//**************************************************************************************************
using System.Xml; 

using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
class PrintEvent : System.Windows.Forms.Form { 
    
    #region 'Windows Form Designer generated code' 
    
    public PrintEvent() { 
        
        
        // This call is required by the Windows Form Designer.
        InitializeComponent(); 
        
        // Add any initialization after the InitializeComponent() call
        
    } 
    
    // Form overrides dispose to clean up the component list.
    protected override void Dispose( bool disposing ) { 
        if ( disposing ) { 
            if ( !( ( components == null ) ) ) { 
                components.Dispose(); 
            } 
        } 
        base.Dispose( disposing ); 
    } 
    
    
    // Required by the Windows Form Designer
    private System.ComponentModel.IContainer components; 
    
    // NOTE: The following procedure is required by the Windows Form Designer
    // It can be modified using the Windows Form Designer.  
    // Do not modify it using the code editor.
    internal System.Windows.Forms.TextBox printXML; 
    internal System.Windows.Forms.Label Label1; 
    internal System.Windows.Forms.Label Label2; 
    [ System.Diagnostics.DebuggerStepThrough() ]

    private void InitializeComponent() { 
		SetApplication();

        this.printXML = new System.Windows.Forms.TextBox(); 
        this.Label1 = new System.Windows.Forms.Label(); 
        this.Label2 = new System.Windows.Forms.Label(); 
        this.SuspendLayout(); 
        // 
        // printXML
        // 
        this.printXML.Location = new System.Drawing.Point( 16, 120 ); 
        this.printXML.Multiline = true; 
        this.printXML.Name = "printXML"; 
        this.printXML.Size = new System.Drawing.Size( 528, 224 ); 
        this.printXML.TabIndex = 2; 
        this.printXML.Text = ""; 
        // 
        // Label1
        // 
        this.Label1.Location = new System.Drawing.Point( 16, 16 ); 
        this.Label1.Name = "Label1"; 
        this.Label1.Size = new System.Drawing.Size( 528, 48 ); 
        this.Label1.TabIndex = 3; 
        this.Label1.Text = "This sample will respond to any print event sent from SBO and respont to it. It w" + "il ask the user if to catch it or not and if to present it in SBO or not."; 
        // 
        // Label2
        // 
        this.Label2.Location = new System.Drawing.Point( 16, 88 ); 
        this.Label2.Name = "Label2"; 
        this.Label2.Size = new System.Drawing.Size( 392, 23 ); 
        this.Label2.TabIndex = 4; 
        this.Label2.Text = "A report that was caught will be displayed here"; 
        // 
        // PrintEvent
        // 
        this.AutoScaleBaseSize = new System.Drawing.Size( 5, 13 ); 
        this.ClientSize = new System.Drawing.Size( 562, 367 ); 
        this.Controls.Add( this.Label2 ); 
        this.Controls.Add( this.Label1 ); 
        this.Controls.Add( this.printXML ); 
        this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle; 
        this.MaximizeBox = false; 
        this.Name = "PrintEvent"; 
        this.Text = "Print Event"; 
        this.ResumeLayout( false ); 
        
		// events handled by SBO_Application_ReportDataEvent
         
        SBO_Application.ReportDataEvent += new SAPbouiCOM._IApplicationEvents_ReportDataEventEventHandler( SBO_Application_ReportDataEvent ); 
        // events handled by SBO_Application_PrintEvent
         
        SBO_Application.PrintEvent += new SAPbouiCOM._IApplicationEvents_PrintEventEventHandler( SBO_Application_PrintEvent ); 
        // events handled by PrintEvent_Load
         
        base.Load += new System.EventHandler( PrintEvent_Load ); 
    } 
    
    
    #endregion 
    
    private SAPbouiCOM.Application SBO_Application; 
    private bool IsRegistered; 
    
    private void SetApplication() { 
        
        // *******************************************************************
        // Use an SboGuiApi object to establish the connection
        // with the application and rturn an initialized appliction object
        // *******************************************************************
        
        SAPbouiCOM.SboGuiApi SboGuiApi = null; 
        string sConnectionString = null; 
        
        SboGuiApi = new SAPbouiCOM.SboGuiApi(); 
        
        // by following the steps specified above, the following
        // statment should be suficient for either development or run mode
        
        sConnectionString = System.Convert.ToString( Environment.GetCommandLineArgs().GetValue( 1 ) ); 
        
        // connect to a running SBO Application
        
        SboGuiApi.Connect( sConnectionString ); 
        
        // get an initialized application object
        
        SBO_Application = SboGuiApi.GetApplication( -1 ); 
        
    } 
    
    
    private void SBO_Application_ReportDataEvent( ref SAPbouiCOM.ReportDataInfo reportDataInfo, out bool BubbleEvent ) { 
		BubbleEvent = true;
        
        if ( reportDataInfo.BeforeAction == true ) { // before action
            int ans = 0; 
            ans = System.Convert.ToInt32( Interaction.MsgBox( "A Report event was sent, do you wish to register to it?", MsgBoxStyle.YesNo, null ) ); 
            IsRegistered = false; 
            
            if ( ans == System.Convert.ToDouble( Constants.vbYes ) ) { 
                reportDataInfo.RegisterForReport( true ); 
                IsRegistered = true; 
            } 
        } 
        
        if ( reportDataInfo.BeforeAction == false ) { // after action
            Interaction.MsgBox( "ReportDataEvent after", (Microsoft.VisualBasic.MsgBoxStyle)(0), null ); 
            if ( IsRegistered ) { 
                if ( ( reportDataInfo.GetPageCount() >= 1 ) ) { 
                    printXML.Text = reportDataInfo.GetReportData( 1, reportDataInfo.GetPageCount(), false ); 
                } 
            } 
        } 
    } 
    
    
    private void SBO_Application_PrintEvent( ref SAPbouiCOM.PrintEventInfo printeventInfo, out bool BubbleEvent ) { 
		BubbleEvent = true;
        if ( printeventInfo.BeforeAction == true ) { // before action
            int ans = 0; 
            ans = System.Convert.ToInt32( Interaction.MsgBox( "Do you want to view it in SBO?", MsgBoxStyle.YesNo, null ) ); 
            
            if ( ans == System.Convert.ToDouble( Constants.vbNo ) ) { 
                BubbleEvent = false; 
            } 
            
        } 
        
        if ( printeventInfo.BeforeAction == false ) { // after action
            Interaction.MsgBox( "PrintEvent after", (Microsoft.VisualBasic.MsgBoxStyle)(0), null ); 
        } 
        
    } 
    
    
    private void PrintEvent_Load( System.Object sender, System.EventArgs e ) { 

		SBO_Application.MessageBox ("Print Event Addon is active.",1,"","","");
        
    } 
    
    
    [STAThread]
    public static void Main() { Application.Run( new PrintEvent() ); }
} 

File: Desarrollos AddOn SAP B1/Samples/COM UI/VB.NET/20.PrintEvent/PrintEvent.vb
'//  SAP MANAGE UI API 2007 SDK Sample
'//****************************************************************************
'//
'//  File:      PrintEvent.vb
'//
'//  Copyright (c) SAP MANAGE
'//
'// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
'// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
'// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'// PARTICULAR PURPOSE.
'//
'// BEFORE STARTING:
'// 1. Add reference to the "SAP Business One UI API"
'// 2. Insert the development connection string to the "Command line argument"
'//-----------------------------------------------------------------
'// 1.
'//    a. Project->Add Reference...
'//    b. select the "SAP Business One UI API 2007" From the COM folder
'//
'// 2.
'//     a. Project->Properties...
'//     b. choose Configuration Properties folder (place the arrow on Debugging)
'//     c. place the following connection string in the 'Command line arguments' field
'// 0030002C0030002C00530041005000420044005F00440061007400650076002C0050004C006F006D0056004900490056
'//
'//**************************************************************************************************
Option Strict Off
Option Explicit On 
Imports System.Xml

Public Class PrintEvent
    Inherits System.Windows.Forms.Form

#Region " Windows Form Designer generated code "

    Public Sub New()
        MyBase.New()

        'This call is required by the Windows Form Designer.
        InitializeComponent()

        'Add any initialization after the InitializeComponent() call

    End Sub

    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    Friend WithEvents printXML As System.Windows.Forms.TextBox
    Friend WithEvents Label1 As System.Windows.Forms.Label
    Friend WithEvents Label2 As System.Windows.Forms.Label
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.printXML = New System.Windows.Forms.TextBox
        Me.Label1 = New System.Windows.Forms.Label
        Me.Label2 = New System.Windows.Forms.Label
        Me.SuspendLayout()
        '
        'printXML
        '
        Me.printXML.Location = New System.Drawing.Point(16, 120)
        Me.printXML.Multiline = True
        Me.printXML.Name = "printXML"
        Me.printXML.ScrollBars = System.Windows.Forms.ScrollBars.Both
        Me.printXML.Size = New System.Drawing.Size(528, 224)
        Me.printXML.TabIndex = 2
        Me.printXML.Text = ""
        '
        'Label1
        '
        Me.Label1.Location = New System.Drawing.Point(16, 16)
        Me.Label1.Name = "Label1"
        Me.Label1.Size = New System.Drawing.Size(528, 48)
        Me.Label1.TabIndex = 3
        Me.Label1.Text = "This sample will respond to any print event sent from SBO and respont to it. It w" & _
        "il ask the user if to catch it or not and if to present it in SBO or not."
        '
        'Label2
        '
        Me.Label2.Location = New System.Drawing.Point(16, 88)
        Me.Label2.Name = "Label2"
        Me.Label2.Size = New System.Drawing.Size(392, 23)
        Me.Label2.TabIndex = 4
        Me.Label2.Text = "A report that was caught will be displayed here"
        '
        'PrintEvent
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(562, 367)
        Me.Controls.Add(Me.Label2)
        Me.Controls.Add(Me.Label1)
        Me.Controls.Add(Me.printXML)
        Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle
        Me.MaximizeBox = False
        Me.Name = "PrintEvent"
        Me.Text = "Print Event"
        Me.ResumeLayout(False)

    End Sub

#End Region

    Private WithEvents SBO_Application As SAPbouiCOM.Application
    Private IsRegistered As Boolean

    Private Sub SetApplication()

        '*******************************************************************
        '// Use an SboGuiApi object to establish the connection
        '// with the application and rturn an initialized appliction object
        '*******************************************************************

        Dim SboGuiApi As SAPbouiCOM.SboGuiApi
        Dim sConnectionString As String

        SboGuiApi = New SAPbouiCOM.SboGuiApi

        '// by following the steps specified above, the following
        '// statment should be suficient for either development or run mode

        sConnectionString = Environment.GetCommandLineArgs.GetValue(1)

        '// connect to a running SBO Application

        SboGuiApi.Connect(sConnectionString)

        '// get an initialized application object

        SBO_Application = SboGuiApi.GetApplication()

    End Sub

    Private Sub SBO_Application_ReportDataEvent(ByRef reportDataInfo As SAPbouiCOM.ReportDataInfo, ByRef BubbleEvent As Boolean) Handles SBO_Application.ReportDataEvent

        If reportDataInfo.BeforeAction = True Then      'before action
            Dim ans As Integer
            ans = MsgBox("A Report event was sent, do you wish to register to it?", MsgBoxStyle.YesNo)
            IsRegistered = False

            If ans = vbYes Then
                reportDataInfo.RegisterForReport(True)
                IsRegistered = True
            End If
        End If

        If reportDataInfo.BeforeAction = False Then     'after action
            MsgBox("ReportDataEvent after")
            If IsRegistered Then
                If (reportDataInfo.GetPageCount() >= 1) Then
                    printXML.Text = reportDataInfo.GetReportData(1, reportDataInfo.GetPageCount())
                End If
            End If
        End If
    End Sub

    Private Sub SBO_Application_PrintEvent(ByRef printeventInfo As SAPbouiCOM.PrintEventInfo, ByRef BubbleEvent As Boolean) Handles SBO_Application.PrintEvent
        If printeventInfo.BeforeAction = True Then      'before action
            Dim ans As Integer
            ans = MsgBox("Do you want to view it in SBO?", MsgBoxStyle.YesNo)

            If ans = vbNo Then
                BubbleEvent = False
            End If

        End If

        If printeventInfo.BeforeAction = False Then      'after action
            MsgBox("PrintEvent after")
        End If

    End Sub

    Private Sub PrintEvent_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        SetApplication()

    End Sub

End Class
